pkg/operator/baremetal_config.go

```diff
@@ -65,3 +77,88 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		ProvisioningDHCPRange:    provisioningDHCPRange,
 	}, nil
 }
+
+func getProvisioningIPCIDR(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningNetworkCIDR != "" && baremetalConfig.ProvisioningIp != "" {
+		_, net, err := net.ParseCIDR(baremetalConfig.ProvisioningNetworkCIDR)
+		if err == nil {
+			cidr, _ := net.Mask.Size()
+			generatedConfig := fmt.Sprintf("%s/%d", baremetalConfig.ProvisioningIp, cidr)
+			return &generatedConfig
+		}
+	}
+	return nil
+}
+
+func getDeployKernelUrl(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalHttpPort), baremetalKernelUrlSubPath)
+		return &generatedConfig
+	}
+	return nil
+}
+
+func getDeployRamdiskUrl(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalHttpPort), baremetalRamdiskUrlSubPath)
+		return &generatedConfig
+	}
+	return nil
+}
+
+func getIronicEndpoint(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalIronicPort), baremetalIronicEndpointSubpath)
+		return &generatedConfig
+	}
+	return nil
+}
+
+func getIronicInspectorEndpoint(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalIronicInspectorPort), baremetalIronicEndpointSubpath)
+		return &generatedConfig
+	}
+	return nil
+}
+
+func getProvisioningDHCPRange(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningDHCPRange != "" {
+		return &(baremetalConfig.ProvisioningDHCPRange)
+	}
+	return nil
+}
+
+func getProvisioningInterface(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningInterface != "" {
+		return &(baremetalConfig.ProvisioningInterface)
+	}
+	return nil
+}
+
+func getMetal3DeploymentConfig(name string, baremetalConfig BaremetalProvisioningConfig) *string {
+	configValue := ""
+	switch name {
+	case "CACHEURL":
+		configValue = ""
```

Comment:
this and line 158 is a bit odd, you could https://github.com/kubernetes/utils/blob/master/pointer/pointer.go#L74

Reply:
Cache URL is soon to be obsolete. Will be updating in a future PR.